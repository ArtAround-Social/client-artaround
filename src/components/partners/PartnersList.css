ul {
  list-style-type: none;
  margin-left: -40px;
  margin-top: -16px;
}

.partners {
  overflow: hidden;
  background-color: rgba(9, 38, 136, 0.822);
  color: white;
  border-radius: 5px;
  text-shadow: none
}



.item {
  border: solid;
  padding: 3px;
  height: 25px;
  border-color: rgba(160, 159, 175, 0.445);
  border-radius: 3px;
  margin-bottom: 4px;
  background-color:  rgb(252, 252, 252);
}

a:link, a:visited {
  background-color: #080355d0;
  color: white;
  padding: 4px 5px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  border-radius: 4px;
}

a:hover, a:active {
  background-color:  #080355fa;
}

.profile {
  float: right;
}