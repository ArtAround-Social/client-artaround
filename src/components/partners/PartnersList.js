import React from 'react';
import PropTypes from 'prop-types';
import PartnerItem from './PartnerItem';

function PartnersList({ partners }) {
  const partnersElement = partners.map(partner => (
    <li key={partner.id}>
      <PartnerItem partner={partner} />
    </li>
  ));

  return(
    <ul>{partnersElement}</ul>
  );
}

PartnersList.propTypes = {
  partners: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired
  }))
};

export default PartnersList;
