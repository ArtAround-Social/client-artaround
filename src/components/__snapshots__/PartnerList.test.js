import React from 'react';
import { shallow } from 'enzyme';
import PartnersList from '../partners/PartnersList';

describe('PartnersList component', () => {
  it('renders partners list', () => {
    const wrapper = shallow(
      <PartnersList
        partners = { [
          { id: 12434, name: 'pavel artist' },
          { id: 43241, name: 'jon artists' }
        ] }
        goToPofile = {() => { }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
