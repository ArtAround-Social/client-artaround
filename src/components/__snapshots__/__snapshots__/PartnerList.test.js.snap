// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnersList component renders partners list 1`] = `
<ul>
  <li
    key="12434"
  >
    <PartnerItem
      partner={
        Object {
          "id": 12434,
          "name": "pavel artist",
        }
      }
    />
  </li>
  <li
    key="43241"
  >
    <PartnerItem
      partner={
        Object {
          "id": 43241,
          "name": "jon artists",
        }
      }
    />
  </li>
</ul>
`;
